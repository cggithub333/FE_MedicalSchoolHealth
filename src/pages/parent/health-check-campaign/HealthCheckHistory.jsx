import FloatingFilterBar from "../../../components/parent/FloatingFilterBar";

const HealthCheckHistory = () => {

  const children = ['Anna Nguyen', 'Liam Tran', 'Sophie Le'];
  const years = ['2023', '2024', '2025'];

  return (
    <div>
      {/* Other content */}
      <FloatingFilterBar childrenList={children} yearList={years} />
    </div>
  );
}

export default HealthCheckHistory;
