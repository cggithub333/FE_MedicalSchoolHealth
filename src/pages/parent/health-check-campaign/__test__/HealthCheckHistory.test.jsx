
import React from 'react';
import { render, screen } from '@testing-library/react';
import HealthCheckHistory from '../HealthCheckHistory';

// Mock child components to isolate HealthCheckHistory
jest.mock('../../../../components/parent/FloatingFilterBar', () => () => <div data-testid="FloatingFilterBar" />);
jest.mock('../../../../components/magic/Breadcrumb/CustomBreadcrumb', () => ({ breadcrumbPairs }) => <div data-testid="Breadcrumb">{breadcrumbPairs && breadcrumbPairs[0]?.title}</div>);
jest.mock('../../../../components/magic/CustomTittle/CustomTitle', () => ({ title }) => <div data-testid="CustomTittle">{title}</div>);
jest.mock('../../../../components/parent/HealthCheckCampaign/HistoryByPupilBySchoolYear', () => () => <div data-testid="HistoryByPupilBySchoolYear" />);

describe('HealthCheckHistory page', () => {
  it('renders all main sections and child components', () => {
    render(<HealthCheckHistory />);
    expect(screen.getByTestId('Breadcrumb')).toBeInTheDocument();
    expect(screen.getByTestId('CustomTittle')).toHaveTextContent('Health Check History');
    expect(screen.getByTestId('HistoryByPupilBySchoolYear')).toBeInTheDocument();
  });

  it('renders the correct breadcrumb titles', () => {
    render(<HealthCheckHistory />);
    expect(screen.getByTestId('Breadcrumb')).toHaveTextContent('Dashboard');
  });

  it('applies the correct styles to the root div', () => {
    const { container } = render(<HealthCheckHistory />);
    const rootDiv = container.firstChild;
    expect(rootDiv).toHaveStyle('background: #e6f8f9');
    expect(rootDiv).toHaveStyle('padding-top: 20px');
    expect(rootDiv).toHaveStyle('padding-bottom: 100px');
  });

  it('renders ThemeProvider and CssBaseline', () => {
    render(<HealthCheckHistory />);
    // CssBaseline injects a <style> tag, so we can check for it
    const styleTags = document.querySelectorAll('style');
    expect(styleTags.length).toBeGreaterThan(0);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<HealthCheckHistory />);
    expect(asFragment()).toMatchSnapshot();
  });
});
