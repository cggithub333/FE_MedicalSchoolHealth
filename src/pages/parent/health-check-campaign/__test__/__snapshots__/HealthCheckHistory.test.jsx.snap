// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HealthCheckHistory page matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="background: rgb(230, 248, 249); height: 100vh; padding-top: 20px; padding-bottom: 100px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-d64g0g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 css-1ml5fja-MuiGrid-root"
      >
        <div
          data-testid="Breadcrumb"
        >
          Dashboard
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-d64g0g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 css-hu5rp3-MuiGrid-root"
      >
        <div
          data-testid="CustomTittle"
        >
          Health Check History
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-thaiby-MuiGrid-root"
    >
      <div
        data-testid="HistoryByPupilBySchoolYear"
      />
    </div>
  </div>
</DocumentFragment>
`;
